### PR TITLE
Top level manifest and dropping submodule

### PR DESCRIPTION
**Pull request by jdmaguire**
This is the latest in a series of improving builds.

The sparking change to this is to remove the git submodule of pravega-rust-client so that we can use cargo-chef & a top-level Cargo manifest. This will help use to remove some redundant compiling (since the different comps have overlapping dependencies).

On my computer, the total build time time goes from 7600s to 3100s. There are still improvements to be made but I'm a bit jank still with Rust building in a container. Those will come with time.

A side-effect of putting in a top level manifest is that all the components of this get one lockfile. All but one comp was using an old version of the glib Rust binding library. I had to update it.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[workspace]
+members = [
+    "apps",
+    "deepstream/pravega_protocol_adapter",
+    "gst-plugin-pravega",
+    "pravega-video",
+    "pravega-video-server",
+    "integration-test"
```
Comment (claudiofahey) [inline Cargo.toml:8]: Keep this list in alphabetical order.

File: integration-test/Cargo.toml [hunk 2]
```diff
@@ -31,9 +31,10 @@ gst-rtsp-server = { package = "gstreamer-rtsp-server", git = "https://gitlab.fre
 java-properties = "1.2"
 lazy_static = "1.4"
 once_cell = "1.7.2"
-pravega-client = { path = "../pravega-client-rust" }
-pravega-client-config = { path = "../pravega-client-rust/config"}
-pravega-client-shared = { path = "../pravega-client-rust/shared"}
+pravega-client = { git = "https://github.com/pravega/pravega-client-rust", rev = "94a435111ae93cdef22e3afb3fb2cbe0dc32ba79" }
+pravega-client-config = { git = "https://github.com/pravega/pravega-client-rust", package = "pravega-client-config", rev = "94a435111ae93cdef22e3afb3fb2cbe0dc32ba79" }
+pravega-client-shared = { git = "https://github.com/pravega/pravega-client-rust", package = "pravega-client-shared", rev = "94a435111ae93cdef22e3afb3fb2cbe0dc32ba79" }
+
```
Comment (claudiofahey) [inline integration-test/Cargo.toml:37]: Remove extra space.

File: gst-plugin-pravega/LICENSE [hunk 3]
```diff
@@ -0,0 +1 @@
+../LICENSE
```
Comment (claudiofahey) [inline gst-plugin-pravega/LICENSE:1]: What is this file needed for?

**Review by claudiofahey — CHANGES_REQUESTED** [review]
Looks good. I'm waiting for the build to complete. In the mean time, there are some very minor changes I would like to see.

File: deepstream/pravega-dev.Dockerfile [hunk 4]
```diff
@@ -53,27 +53,39 @@ RUN set -eux; \
...
+COPY --from=planner /usr/src/gstreamer-pravega/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+
+FROM builder-base as final
 ## Build gst-plugin-pravega.
 
+# Copy over the cached dependencies
+COPY --from=cacher /usr/src/gstreamer-pravega/target target
+COPY --from=cacher /usr/local/cargo /usr/local/cargo
+COPY . .
```
Comment (claudiofahey) [inline deepstream/pravega-dev.Dockerfile:72]: For the future, we should be sure to add everything possible to .dockerignore to avoid invalidating the cache unnecessarily.

File: Cargo.toml (hunk 1 above)
Comment (jdmaguire) [reply Cargo.toml:8]: My brain keeps wanting to put these in dependency order! 🤣

**Review by claudiofahey — APPROVED** [review]
